src/site/topic24.rst

```diff
@@ -268,6 +268,130 @@ Remove Minimum & Maximum
 General Remove
 ==============
 
+* This is probably the most complex functionality we will discuss
+* To help, the discussion will be broken up
+
+.. code-block:: java
+    :linenos:
+
+    public T remove(T element) {
+        T returnElement = null;
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        } else if (root.getData().equals(element)) {
+            returnElement = root.getData();
+            root = findReplacementNode(root);
```

Comment:
I can't help but feel like there's a way to move this logic the private `remove`. Logic is duplicated here.

Reply:
Brain end day condition. Seems similar to #347